### PR TITLE
MSVC assembly fixups/cleanups

### PR DESCRIPTION
**Pull request by TheComputerGuy96**
This is a separate MR based on a suggestion in https://github.com/CookiePLMonster/SilentPatch/pull/95#issuecomment-2470023367

### CODE REVIEW

File: SilentPatchIII/SilentPatchIII.cpp [hunk 1]
```diff
@@ -781,7 +782,7 @@ namespace GenerateNewPickup_ReuseObjectFix
 			mov		edi, eax
 
 			push	edi
-			call	[WorldRemove]
+			call	offset WorldRemove
```
Comment (CookiePLMonster) [inline SilentPatchIII/SilentPatchIII.cpp]: This is probably wrong.

**Review by CookiePLMonster — CHANGES_REQUESTED** [review]

File: SilentPatchIII/SilentPatchIII.cpp (hunk 1 above)
Comment (TheComputerGuy96) [reply SilentPatchIII/SilentPatchIII.cpp]: That's what syntax I'm using in the GCC inline assembly statement (I can change that if MASM complains)

Comment (CookiePLMonster) [reply SilentPatchIII/SilentPatchIII.cpp]: I mean that the call is supposed to dereference `WorldRemove` and call that, `call offset` will probably attempt to call the address of this variable.

Comment (TheComputerGuy96) [reply SilentPatchIII/SilentPatchIII.cpp]: That `offset` keyword introduces run-time issues in GTA 3 (I got a segfault in a user callback on Wine) so I'm definitely removing it

**Comment by CookiePLMonster** [comment]
Looks good overall, I'll give it a spin next time I am working on SP.

**Review by CookiePLMonster — APPROVED** [review]
Tested all 3 games, looks all good now. Is this fine to squash into a single commit?

**Comment by TheComputerGuy96** [comment]
@CookiePLMonster The SA part should probably be left unsquashed (because it's relatively big)

**Comment by CookiePLMonster** [comment]
OK, I see the review fix was force pushed, so even unsquashed history is clean. Fine by me then.

**Pull request merged**